### PR TITLE
cod: fix tests

### PR DESCRIPTION
**Pull request by superherointj**
cod: fix tests

Co-authored-by: @jnetod @thiagokokada

```
@nix { "action": "setPhase", "phase": "checkPhase" }
running tests
/build/source/test/binaries /build/source
patching script interpreter paths in argparse-subcommand.py
argparse-subcommand.py: interpreter directive changed from "#!/usr/bin/env python3" to "/nix/store/9px00aaqzb6n5p03i9wd8rx3msg95y9r-python3-3.9.11/bin/python3"
patching script interpreter paths in cat.py
cat.py: interpreter directive changed from "#!/usr/bin/env python3" to "/nix/store/9px00aaqzb6n5p03i9wd8rx3msg95y9r-python3-3.9.11/bin/python3"
patching script interpreter paths in default-subcommand.py
default-subcommand.py: interpreter directive changed from "#!/usr/bin/env python3" to "/nix/store/9px00aaqzb6n5p03i9wd8rx3msg95y9r-python3-3.9.11/bin/python3"
patching script interpreter paths in foo_v1.py
foo_v1.py: interpreter directive changed from "#!/usr/bin/env python3" to "/nix/store/9px00aaqzb6n5p03i9wd8rx3msg95y9r-python3-3.9.11/bin/python3"
patching script interpreter paths in foo_v2.py
foo_v2.py: interpreter directive changed from "#!/usr/bin/env python3" to "/nix/store/9px00aaqzb6n5p03i9wd8rx3msg95y9r-python3-3.9.11/bin/python3"
patching script interpreter paths in naval-fate.py
naval-fate.py: interpreter directive changed from "#!/usr/bin/env python3" to "/nix/store/9px00aaqzb6n5p03i9wd8rx3msg95y9r-python3-3.9.11/bin/python3"
/build/source
=== RUN   TestCanonizeExecutablePath
--- PASS: TestCanonizeExecutablePath (0.00s)
=== RUN   TestIsCommandMatchingContext
--- PASS: TestIsCommandMatchingContext (0.00s)
=== RUN   TestNewSqliteStorage
--- PASS: TestNewSqliteStorage (0.00s)
=== RUN   TestCRUD
--- PASS: TestCRUD (0.00s)
=== RUN   TestAddSamePage
--- PASS: TestAddSamePage (0.00s)
=== RUN   TestListCommands
--- PASS: TestListCommands (0.00s)
PASS
ok  	github.com/dim-an/cod/datastore	0.005s
=== RUN   TestParseArgparse
--- PASS: TestParseArgparse (0.00s)
=== RUN   TestParseArgparseContext
--- PASS: TestParseArgparseContext (0.00s)
=== RUN   TestParseDocker
--- PASS: TestParseDocker (0.00s)
=== RUN   TestParseUsageSubCommand
--- PASS: TestParseUsageSubCommand (0.00s)
=== RUN   TestIsJavaStyleFlag
--- PASS: TestIsJavaStyleFlag (0.00s)
=== RUN   TestParseCatHelp
2022/04/07 14:55:15 Parser argparse failed with error cannot find usage
--- PASS: TestParseCatHelp (0.00s)
=== RUN   TestParseQuWriteFileHelp
--- PASS: TestParseQuWriteFileHelp (0.00s)
=== RUN   TestParseLsHelp
2022/04/07 14:55:15 Parser argparse failed with error cannot find usage
--- PASS: TestParseLsHelp (0.00s)
=== RUN   TestMultipleFlagOccurrences
2022/04/07 14:55:15 Parser argparse failed with error error parsing usage: cannot tokenize: <flags>
--- PASS: TestMultipleFlagOccurrences (0.00s)
=== RUN   TestJavaStyleInclusion
2022/04/07 14:55:15 Parser argparse failed with error error parsing usage: cannot tokenize: <flags>
--- PASS: TestJavaStyleInclusion (0.00s)
=== RUN   TestJavaStyle
2022/04/07 14:55:15 Parser argparse failed with error error parsing usage: cannot tokenize: <flags>
--- PASS: TestJavaStyle (0.00s)
=== RUN   TestOptionInVeryBeginningOfLine
2022/04/07 14:55:15 Parser argparse failed with error error parsing usage: cannot tokenize: <flags>
--- PASS: TestOptionInVeryBeginningOfLine (0.00s)
=== RUN   TestFindIndentedParagraph
--- PASS: TestFindIndentedParagraph (0.00s)
PASS
ok  	github.com/dim-an/cod/parse_doc	0.003s
=== RUN   TestQuote
--- PASS: TestQuote (0.00s)
=== RUN   TestBashRemoveCompletions
--- PASS: TestBashRemoveCompletions (0.00s)
=== RUN   TestTokenizeStrings
--- PASS: TestTokenizeStrings (0.00s)
PASS
ok  	github.com/dim-an/cod/shells	0.002s
=== RUN   TestExampleConfiguration
--- PASS: TestExampleConfiguration (0.03s)
=== RUN   TestLearn
--- PASS: TestLearn (0.02s)
=== RUN   TestLearnBroken
--- PASS: TestLearnBroken (0.01s)
=== RUN   TestLearnUpdateShorter
--- PASS: TestLearnUpdateShorter (0.04s)
=== RUN   TestLearnFromPATH
--- PASS: TestLearnFromPATH (0.02s)
=== RUN   TestMergeLearn
--- PASS: TestMergeLearn (0.06s)
=== RUN   TestLearnArgparseSubCommand
--- PASS: TestLearnArgparseSubCommand (0.09s)
=== RUN   TestLearnDefaultSubCommand
--- PASS: TestLearnDefaultSubCommand (0.06s)
=== RUN   TestLearnList
--- PASS: TestLearnList (0.04s)
=== RUN   TestLearnListRemove
--- PASS: TestLearnListRemove (0.05s)
=== RUN   TestLearnListBasenameSelector
--- PASS: TestLearnListBasenameSelector (0.04s)
=== RUN   TestLearnListPathSelector
--- PASS: TestLearnListPathSelector (0.04s)
=== RUN   TestLearnListIdSelector
--- PASS: TestLearnListIdSelector (0.04s)
=== RUN   TestPollUpdates
--- PASS: TestPollUpdates (0.02s)
=== RUN   TestDaemonSmoke
--- PASS: TestDaemonSmoke (0.00s)
=== RUN   TestAttachDetach
2022/04/07 14:55:16 killing shell: 10540
--- PASS: TestAttachDetach (0.31s)
=== RUN   TestUpdateReplace
--- PASS: TestUpdateReplace (0.04s)
=== RUN   TestUpdateMerge
--- PASS: TestUpdateMerge (0.07s)
=== RUN   TestUpdateNoChange
--- PASS: TestUpdateNoChange (0.04s)
=== RUN   TestUpdateBroken
--- PASS: TestUpdateBroken (0.03s)
PASS
ok  	github.com/dim-an/cod/test	1.051s
=== RUN   TestGlobExpression_Match
--- PASS: TestGlobExpression_Match (0.00s)
PASS
ok  	github.com/dim-an/cod/util	0.001s
```

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by thiagokokada — CHANGES_REQUESTED** [review]
Please fill up the pull request template.

File: pkgs/tools/misc/cod/default.nix [hunk 1]
```diff
@@ -15,7 +15,16 @@ buildGoModule rec {
 
   ldflags = [ "-s" "-w" "-X main.GitSha=${src.rev}" ];
 
-  doCheck = false;
+  nativeBuildInputs = [ python3 ];
```
Comment (jnetod) [inline pkgs/tools/misc/cod/default.nix:18]: If it's a `checkPhase`-only dependency, better to add it to `checkInputs` because semantics.

**Review by thiagokokada — APPROVED** [review]

**Comment by zendo** [comment]
Build failed.

`2022/07/10 01:06:43 killing shell: 8223
--- PASS: TestAttachDetach (0.31s)
=== RUN   TestUpdateReplace
--- PASS: TestUpdateReplace (0.05s)
=== RUN   TestUpdateMerge
--- PASS: TestUpdateMerge (0.08s)
=== RUN   TestUpdateNoChange
--- PASS: TestUpdateNoChange (0.05s)
=== RUN   TestUpdateBroken
--- PASS: TestUpdateBroken (0.03s)
FAIL
FAIL    github.com/dim-an/cod/test      1.179s
FAIL`

**Comment by superherointj** [comment]
Notice, when I proposed PR tests were green:
`--- PASS: TestLearnArgparseSubCommand`

Now it is failing:
`--- FAIL: TestLearnArgparseSubCommand`

Reviewers tested too. Question is WHY?

`TestLearnArgparseSubCommand` unit test could be skipped, but because of the sudden change it could be something else causing this.

**Comment by superherointj** [comment]
@zendo Please review #180908

**Pull request merged**